components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -144,7 +153,22 @@ void run_scalar_valued_fns()
   psl=PF::calculate_psl( T_ground , p_ground, phi_ground );
   REQUIRE( Check::approx_equal( psl_exact, psl, test_tol) );
   REQUIRE( psl<p_ground );
-} 
+
+  // Get dx from grid cell area property tests:
+  RealType area, lat, grid_dx;
+  area = 0.0; lat = 1.0;
+  REQUIRE( Check::equal(PF::calculate_dx_from_area(area,lat),0.0) );
+  area = (pi/180.0)*(pi/180.0); lat = 0.0;
+  REQUIRE( Check::equal(PF::calculate_dx_from_area(area,lat), coeff_1-coeff_2+coeff_3) );
+  lat = pi/2.0;
+  REQUIRE( Check::equal(PF::calculate_dx_from_area(area,lat), coeff_1+coeff_2+coeff_3) );
+  
+  lat     = pdf_lat(engine);
+  area    = pdf_area(engine);
+  grid_dx = (coeff_1 - coeff_2 * std::cos(2.0*lat) + coeff_3 * std::cos(4.0*lat)) * std::sqrt(area)*(180.0/pi);
+  REQUIRE( Check::equal(PF::calculate_dx_from_area(area,lat),grid_dx) ); 
```

Comment:
Imho, this is not a real property test, in that we are spelling out the very same formula we have in the fcn definition (probably via copy/paste as well). This only checks that the formula you wrote here is the same as the formula you wrote there.

Instead, I'd like to check more values of lat: pi/4 and maybe pi/8 for instance. Also check that f(lat)==f(-lat). Another thing could be to check that f(area=4) == 2*f(area=1) (this might require some tol, not sute).

Reply:
excellent suggestion.  I'll put that in.